### PR TITLE
[SPARK-31612][SQL][DOCS][FOLLOW-UP] Fix a few issues in SQL ref

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Fix a few issues in SQL Reference

### Why are the changes needed?
To make SQL Reference look better

### Does this PR introduce _any_ user-facing change?
Yes.
before: 
Screen Shot 2020-05-21 at 11 41 34 PM

after:

Screen Shot 2020-05-21 at 11 41 17 PM

before:
Screen Shot 2020-05-21 at 11 45 22 PM

after:

Screen Shot 2020-05-21 at 11 45 02 PM

before:
Screen Shot 2020-05-21 at 11 41 57 PM

after

Screen Shot 2020-05-21 at 11 42 26 PM

### How was this patch tested?
Manually build and check

### CODE REVIEW

**Comment by gatorsmile** [comment]
Thank you for your quick fix!

**Comment by maropu** [comment]
retest this please

File: docs/sql-ref.md [hunk 1]
```diff
@@ -21,19 +21,19 @@ license: |
...
+   * [User-Defined Aggregate Functions (UDAFs)](sql-ref-functions-udf-aggregate.html)
+   * [Integration with Hive UDFs/UDAFs/UDTFs](sql-ref-functions-udf-hive.html)
  * [Identifiers](sql-ref-identifier.html)
  * [Literals](sql-ref-literals.html)
  * [Null Semanitics](sql-ref-null-semantics.html)
- * [ANSI Compliance](sql-ref-ansi-compliance.html)
  * [SQL Syntax](sql-ref-syntax.html)
    * [DDL Statements](sql-ref-syntax-ddl.html)
-   * [DML Statements](sql-ref-syntax-ddl.html)
+   * [DML Statements](sql-ref-syntax-dml.html)
```
Comment (maropu) [inline docs/sql-ref.md:37]: Ur, nice catch...

**Review by maropu — APPROVED** [review]
cc: @srowen

**Review by srowen — COMMENTED** [review]
Looks fine to me - they weren't in the previous order for a reason? alphabetizing seems fine if that's the standard.

**Comment by huaxingao** [comment]
@srowen This is for 3.0. Thanks!

**Comment by maropu** [comment]
Merged to master/3.0.

**Comment by huaxingao** [comment]
Thanks @maropu @srowen

**Pull request closed without merging**